src/main/java/io/kestra/plugin/influxdb/FluxQuery.java

```diff
@@ -0,0 +1,281 @@
+                    fetchType: FETCH_ONE
+                """
+        )
+    }
+)
+public class FluxQuery extends AbstractTask implements RunnableTask<FluxQuery.Output> {
+    @Schema(
+        title = "Flux query to execute",
+        description = "The Flux query to run against InfluxDB"
+    )
+    @NotNull
+    private Property<String> query;
+
+    @Schema(
+        title = "The way you want to store the data.",
+        description = "FETCH_ONE output the first row, "
+            + "FETCH output all the rows, "
+            + "STORE store all rows in a file, "
+            + "NONE do nothing."
+    )
+    @Builder.Default
+    private Property<FetchType> fetchType = Property.of(FetchType.FETCH);
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        Logger logger = runContext.logger();
+        String renderedQuery = runContext.render(query).as(String.class).orElseThrow();
+        String renderedOrg = runContext.render(org).as(String.class).orElseThrow();
+
+        try (
+            InfluxDBClient client = this.connection.client(runContext);
+        ){
+            QueryApi queryApi = client.getQueryApi();
+            logger.debug("Starting query: {}", query);
+
+            if (bucket != null) {
+                logger.warn("Bucket is ignored for FluxQuery as it's embedded in the query string.");
+            }
+
+            List<FluxTable> tables = queryApi.query(renderedQuery,renderedOrg);
+
+            Output.OutputBuilder outputBuilder = Output.builder();
+            int recordCount = 0;
+
+            for (FluxTable table : tables) {
+                recordCount += table.getRecords().size();
+            }
+
+            switch (runContext.render(fetchType).as(FetchType.class).orElseThrow()) {
+                case FETCH:
+                    Pair<List<Map<String, Object>>, Integer> fetch = this.fetch(tables);
+                    outputBuilder
+                        .rows(fetch.getLeft())
+                        .size(fetch.getRight());
+                    break;
+
+                case FETCH_ONE:
+                    Map<String, Object> row = this.fetchOne(tables);
+                    outputBuilder
+                        .row(row)
+                        .size(row != null ? 1 : 0);
+                    break;
+
+                case STORE:
+                    Pair<URI, Long> store = this.store(runContext, tables);
+                    outputBuilder
+                        .uri(store.getLeft())
+                        .size(store.getRight().intValue());
+                    break;
+
+                case NONE:
+                    outputBuilder.size(recordCount);
+                    break;
+            }
+
+            runContext.metric(Counter.of("records", recordCount));
+            return outputBuilder.total((long) tables.size()).build();
+        }
+    }
+
+    protected Pair<List<Map<String, Object>>, Integer> fetch(List<FluxTable> tables) {
```

Comment:
you can use a new Record with 2 named vars here for sake of clarity

Reply:
like `private record FetchResult(List<Map<String, Object>> results, Integer size){}`